packages/store/storehooks.js

```diff
@@ -0,0 +1,258 @@
+      dispatch(setFirstLoginAfterSignup(stringToBoolean(cookies.get('firstLoginAfterSignup'))))
+    ];
+    const multitenancy = hasMultitenancy || isHosted || isEnterprise;
+    if (multitenancy) {
+      tasks.push(dispatch(getUserOrganization()));
+    }
+    return Promise.all(tasks);
+  }, [dispatch, hasMultitenancy, isHosted, isEnterprise]);
+
+  const interpretAppData = useCallback(() => {
+    let settings = {};
+    if (cookies.get('_ga') && typeof hasTrackingEnabled === 'undefined') {
+      settings.trackingConsentGiven = true;
+    }
+    let tasks = [
+      dispatch(setDeviceListState({ selectedAttributes: columnSelection.map(column => ({ attribute: column.key, scope: column.scope })) })),
+      dispatch(setTooltipsState(tooltips)), // tooltips read state is primarily trusted from the redux store, except on app init - here user settings are the reference
+      dispatch(saveUserSettings(settings))
+    ];
+    // checks if user id is set and if cookie for helptips exists for that user
+    tasks = maybeAddOnboardingTasks({ devicesByStatus, dispatch, tasks, onboardingState });
+
+    if (canManageUsers && intervalUnit && intervalUnit !== timeUnits.days) {
+      const duration = dayjs.duration(interval, intervalUnit);
+      const days = duration.asDays();
+      if (days < 1) {
+        tasks.push(Promise.resolve(setTimeout(() => dispatch(setShowStartupNotification(true)), TIMEOUTS.fiveSeconds)));
+      } else {
+        const roundedDays = Math.max(1, Math.round(days));
+        tasks.push(dispatch(saveGlobalSettings({ offlineThreshold: { interval: roundedDays, intervalUnit: timeUnits.days } })));
+      }
+    }
+
+    // the following is used as a migration and initialization of the stored identity attribute
+    // changing the default device attribute to the first non-deviceId attribute, unless a stored
+    // id attribute setting exists
+    const identityOptions = identityAttributes.filter(attribute => !['id', 'Device ID', 'status'].includes(attribute));
+    if (!id_attribute && identityOptions.length) {
+      tasks.push(dispatch(saveGlobalSettings({ id_attribute: { attribute: identityOptions[0], scope: 'identity' } })));
+    } else if (typeof id_attribute === 'string') {
+      let attribute = id_attribute;
+      if (attribute === 'Device ID') {
+        attribute = 'id';
+      }
+      tasks.push(dispatch(saveGlobalSettings({ id_attribute: { attribute, scope: 'identity' } })));
+    }
+    return Promise.all(tasks);
+  }, [
+    columnSelection,
+    dispatch,
+    identityAttributes,
+    hasTrackingEnabled,
+    canManageUsers,
+    devicesByStatus,
+    id_attribute,
+    interval,
+    intervalUnit,
+    onboardingState,
+    tooltips
+  ]);
+
+  const initializeAppData = useCallback(
+    () =>
+      retrieveAppData()
+        .then(interpretAppData)
+        // this is allowed to fail if no user information are available
+        .catch(err => console.log(extractErrorMessage(err)))
```

Comment:
do we need console.log here or commonErrorHandler?

Reply:
the `commonErrorHandler` would create a snackbar, but here it's ok to get some errors (e.g. when running OS on-prem) - so logging it just in case 🤷‍♂️ - but that we can of course change later on!